docs/en/advanced-concepts/android-testing-with-docker.md

```diff
@@ -0,0 +1,66 @@
+## Android Tests With Docker
+
+When time comes to run appium tests on your own CI (like Jenkins) you would probably think about using docker.
+In order to run Android tests you will also need to have an emulator. Fortunately you can find official docker image that has both Appium and Android Emulator bundled inside: 
+https://hub.docker.com/r/appium/appium-emulator/
+ 
+All you need to know is just 2 simple tricks:
+1. Run ADB server using `-a` switch in the docker container running emulator. ADB will listen to all connected network interfaces, so you can connect emulator from another running container
+2. You will need to setup port forwarding from the container running Appium to the container running emulator. 
+Appium uses a `bootstrap-port` to setup communication to the emulator and it will bind to the localhost by default.
+But since emulator running in another container, you will need to bind that local `bootstrap-port` to the same port on the emulator container
+
+Here is a sample docker compose file that does all these tricks:
+```yaml
+version: "2"
+
+services:
+  emulator:
+    container_name: emulator
+    image: appium/appium-emulator
+    command: >
+          bash -c "adb -a nodaemon server &
+                   emulator64-arm -avd Nexus -no-boot-anim -no-window -noaudio -gpu off"
+
+  appium:
+    container_name: appium
+    image: appium/appium-emulator
```

Comment:
@serggl I didn't understand why are you using ```appium/appium-emulator``` here and what is the idea behind running appium in one container and emulator in another container? Is it to run tests parallely in a single device/emulator? If you have looked at ```appium/appium-emulator``` it runs both appium and android emulator in a single container. Also FYI, if you look at recent release of https://github.com/appium/appium-docker-android/releases/tag/2.0-barcelona, we have deprecated official emulator image in favor of https://github.com/butomo1989/docker-android.  We will be adding a detailed readme links to advanced concepts section once we use docker-appium as a base image in docker-android. Probably you can have a look at [this](https://github.com/butomo1989/docker-android/blob/master/docker-compose.yml) and [this](https://github.com/SrinivasanTarget/DockerAppium/blob/master/docker-compose.yml). 
Ping @butomo1989